libexec/rtld-elf/cheri/cheri_reloc.h

```diff
@@ -115,7 +118,11 @@ process_r_cheri_capability(Obj_Entry *obj, Elf_Word r_symndx,
 			}
 		}
 		/* Remove write permissions and set bounds */
+#ifdef __CHERI_PURE_CAPABILITY__
 		symval = make_function_pointer_with_addend(def, defobj, addend);
+#else
+		symval = make_function_cap_pointer_with_addend(def, defobj, addend);
+#endif
```

Comment:
Maybe we should always have make_function_cap_with_addend (without the unnecessary cap_pointer) etc that are just aliases for the pointer ones on MIPS and RISC-V so there doesn't need to be an `#ifdef` every time you want to use them?